config_converter/config_converter.py

```diff
@@ -0,0 +1,202 @@
+"""Program that converts a fluentd config file to a master agent config file.
+
+Usage: Run the exec file in the bin folder with 2 arguments - the
+path to the config file to migrate, and a directory to store the output
+master agent config file in.
+
+To do:
+1. Improve the cli. Include a --help option. Allow optional args like
+   log-level and log-filepath (see project docs).
+2. Create a mapping for log_level if there in the fluentd config file and input
+   default ones if not given
+3. Generate stats like #fields converted, ignored, and output as schema
+4. Generate logs
+5. Fix types of numbers, lists
+6. Generate tests to check if for right logs, right stats, above changes
+"""
+
+import copy
+import sys
+import os
+import subprocess
+import yaml
+from google.protobuf import json_format
+import config_converter.config_pb2 as config_pb2
+
+
+def read_file(path):
+    """Returns content of file at path."""
+    with open(path, 'rt') as f:
+        return f.read()
+
+
+def get_object():
+    """Run ruby exec file and get message object."""
+    try:
+        subprocess.run(
+            ['config_converter/config_parser_ruby/bin/' + 'config_parser'] +
+            sys.argv[1:],
+            check=True)
+    except:
+        sys.exit()
+    return json_format.Parse(read_file(sys.argv[-1] + '/config.json'),
+                             config_pb2.Directive())
+ 
+
+def write_to_yaml(result):
+    """Writes created result dictionary to a yaml file."""
+    file_name = os.path.splitext(os.path.basename(sys.argv[1]))[0]
+    with open(f'{sys.argv[-1]}/{file_name}.yaml', 'w') as f:
+        yaml.dump(result, f)
+
+
+class ConfigConverter():
+    """Class containing tools to map fluentd fields to master agent fields."""
+    def __init__(self):
+        self.result = {'logs_module': {}}
+        self.create_mapping_info()
+        self.config_obj = get_object()
+        self.extract_root_dirs()
+        write_to_yaml(self.result)
+        subprocess.run(['rm', f'{sys.argv[-1]}/config.json'], check=True)
```

Comment:
consider using argparse for the arguments passed in - so argparse helps you figure out how to parse the arguments, and it also help you document the arguments.

Argparse also allows `FileType` (https://docs.python.org/3/library/argparse.html#filetype-objects) which supports multiple functionalities.

Reply:
I decided not use use argparse in this PR since I already implemented it in the ruby code written earlier, and this code directly uses that. However I will be implementing that in my next PR when I also address the other optional arguments that can be passed in. But you're right - it will be better to parse and document the arguments in this file and I will work on that soon. Hope that's okay!